tests/cases/conformance/types/mapped/mappedTypesGenericTuples.ts

```diff
@@ -0,0 +1,45 @@
+// @strict: true
+// @noEmit: true
+
+// Property keys are `number` following the fixed part of a tuple
+
+type K<T> = { [P in keyof T]: P };
+type M<T> = { [P in keyof T]: T[P] };
+
+type KA = K<[string, string, boolean]>;  // ["0", "1", "2"]
+type KB = K<[string, string, ...string[], string]>;  // ["0", "1", ...number[], number]
+type KC = K<[...string[]]>;  // number[]
+type KD = K<string[]>;  // number[]

```

Comment:
I know it's not allowed to have non-generic variable elements follow a non-generic rest, but it would be good to make sure the checker tolerates states like that

```suggestion
type KD = K<string[]>;  // number[]
type KE = K<[string, ...boolean[], number?]>;
type KE = K<[string, ...boolean[], ...number[]]>;
```

Reply:
Since you bring this up here... this is killing me 😅 
```ts
type A = [number, ...boolean[], ...string[]] // error
type B = [number, ...Array<boolean>, ...Array<string>] // ok
```

Shouldn't those really behave the same way? IIRC I have a PR laying around somewhere that makes this error go away (by matching the behavior of the latter)

Reply:
Is the PR already open?

Not that it's related, but you inspired me to file something else I've put off mentioning for a while: #57034

Reply:
It is probably not the intention that we *want* to allow that code to be honest. But I'll let Anders weigh in.

Reply:
> Is the PR already open?

sure thing :) https://github.com/microsoft/TypeScript/pull/55446

Reply:
@DanielRosenwasser Those extra examples wouldn't really check anything related to this PR since we normalize the tuple type before the instantiation logic even sees it.

Reply:
@Andarist I still think the error makes sense, but we should fix the error reporting logic to be consistent between the notations. Permitting `[...string[], ...number[]]` with no errors will just make users think we can actually check for a sequence of strings followed by a sequence of numbers, but that's not the case.

Reply:
I'd assume that people might already be relying on the fact spreading like this might "normalize" the shape, so I thought that is off the table as it would be considered a breaking change.

Reply:
Error reporting fixed with the latest commit.